pilot/cmd/pilot-agent/main.go

```diff
@@ -279,9 +277,14 @@ var (
 			secOpts.InitialBackoffInMilliSec = int64(initialBackoffInMilliSecEnv)
 			// Disable the secret eviction for istio agent.
 			secOpts.EvictionDuration = 0
-			if citadel.ProvCert != "" {
-				secOpts.AlwaysValidTokenFlag = true
+			secOpts.AlwaysValidTokenFlag = (platform == security.K8S)
+
+			credFetcher, err := credentialfetcher.NewCredFetcher(platform, trustDomain, jwtPath)
+			if err != nil {
+				return fmt.Errorf("failed to create credential fetcher: %v", err)
 			}
+			log.Infof("Start credential fetcher on %s platform in %s trust domain", platform, trustDomain)
```

Comment:
why would we start the credFetcher on all platforms ? 

Please add a plugin or option specifically to start a cred fetcher - and only start it if the option is set.

As a general rule - new features should not be on by default ( and certainly not 'last minute large changes')